DSpace 6 build fails using latest Maven version due to blocking of restlet m2 repository

**Describe the bug**

Using the latest version of maven (3.8.1), building DSpace 6 (or DSpace 5), the build fails with a message like the following (formatted for readability):

```
[ERROR] Failed to execute goal on project dspace-solr: 
  Could not resolve dependencies for project org.dspace:dspace-solr:war:6.4-SNAPSHOT: 
    Failed to collect dependencies at org.apache.solr:solr-core:jar:4.10.4 
      -> org.restlet.jee:org.restlet:jar:2.1.1: 
        Failed to read artifact descriptor for org.restlet.jee:org.restlet:jar:2.1.1: 
          Could not transfer artifact org.restlet.jee:org.restlet:pom:2.1.1 from/to 
          maven-default-http-blocker (http://0.0.0.0/): 
            Blocked mirror for repositories: [
              maven-snapshots (http://oss.sonatype.org/content/repositories/snapshots, default, snapshots), 
              maven-restlet (http://maven.restlet.org, default, releases+snapshots), 
              apache.snapshots (http://repository.apache.org/snapshots, default, disabled)
            ] -> [Help 1]
```

This is failing due to a security measure added to maven 3.8.1 ([see release notes](https://maven.apache.org/docs/3.8.1/release-notes.html)). This measure prevents the use of plain old http maven repositories in order to reduce the chance of man-in-the-middle attacks.

**To Reproduce**

Steps to reproduce the behavior:

1. Clear out your ~/.m2/repository/org/restlet/ directory, if present (this may not be wholly necessary, but it ensures maven will be forced to try to download the library in any case).
2. Make sure you're using the latest version of maven (currently 3.8.1)
3. Try to build dspace...the build should fail with an error like the above.

**Expected behavior**

It should successfully build, as it does with Maven 3.8.0 and below.

--- a/pom.xml
+++ b/pom.xml
@@ -1554,11 +1554,11 @@
         </repository>
     </distributionManagement>
 
-    <!-- Enable access to artifacts in Sonatype's snapshot repo for Snapshots ONLY -->
     <repositories>
+        <!-- Enable access to artifacts in Sonatype's snapshot repo for Snapshots ONLY -->
         <repository>
           <id>maven-snapshots</id>
-          <url>http://oss.sonatype.org/content/repositories/snapshots</url>
+          <url>https://oss.sonatype.org/content/repositories/snapshots</url>
           <layout>default</layout>
             <releases>
               <enabled>false</enabled>
@@ -1566,6 +1566,11 @@
           <snapshots>
             <enabled>true</enabled>
           </snapshots>
+        </repository>
+        <!-- Add mirror for restlet - maven-default-http-blocker fix -->
+        <repository>
+            <id>restlet-mirror</id>
+            <url>https://maven.restlet.talend.com</url>
         </repository>
     </repositories>
 

